### ISSUE TITLE
Can't add accessory to home.

### ISSUE THREAD

**Issue by incipio-swetha** [fredericfalliere/HomeKit#5]
Hi,

I am getting below errors when I am trying to add the accessory. Any help is much appreciated.

![img_0272](https://cloud.githubusercontent.com/assets/8664403/4370723/dad9a0f2-430c-11e4-8ea8-e9477d0f8e52.PNG)
![img_0273](https://cloud.githubusercontent.com/assets/8664403/4370722/dad8fbfc-430c-11e4-803c-6957e0effe22.PNG)
![img_0274](https://cloud.githubusercontent.com/assets/8664403/4370724/dade0372-430c-11e4-9e23-351eda42b809.PNG)

Thanks,
Swetha.

**Comment by fredericfalliere** [comment]
do you have provisionning profiles with homekit checked?

**Comment by incipio-swetha** [comment]
Yes, its checked in the provisioning profile. I am getting a error while calling the below code:

homeManager.primaryHome.addAccessory(accessories[indexPath.row] as HMaccessory, completionHandler: ({(error:NSError!) in

The above code is giving the below error when I debug:
fatal error: unexpectedly found nil while unwrapping an Optional value

Thanks,
Swetha.

**Comment by incipio-swetha** [comment]
I enabled the home kit in provisioning profile, but I am still getting the below error:
error: Error Domain=HMErrorDomain Code=2 "The operation couldn’t be completed. (HMErrorDomain error 2.)".

Any help is appreciated.

![screen shot 2014-09-23 at 5 46 33 pm](https://cloud.githubusercontent.com/assets/8664403/4371713/8cc10a68-431b-11e4-980a-fc0ffa223bd1.png)

**Comment by fredericfalliere** [comment]
Is it a real accessory or ont from the simulator ?

If it's a real one, maybe your accessory as some problems. Try downloading the homekit accessory controller to make sure your accessory works with homekit.

**Comment by incipio-swetha** [comment]
Thank u, its a real accessory. The accessory which I created using "Homekit accessory simulator" is working fine. The real accessory is not working. 

Thanks for your update, can you please let em know if a accessory needs a certain characteristic to be identified.

**Comment by incipio-swetha** [comment]
Also from when can I download "homekit accessory controller"?
